File: libgomp/testsuite/libgomp.fortran/cancel-taskgroup-2.f90
! { dg-do run }
! { dg-set-target-env-var OMP_CANCELLATION "true" }

  use omp_lib
  integer :: i

  !$omp parallel
    !$omp taskgroup
      !$omp task
	!$omp cancel taskgroup
	call abort
      !$omp endtask
    !$omp endtaskgroup
  !$omp endparallel
  !$omp parallel private (i)
    !$omp barrier
    !$omp single
      !$omp taskgroup
	do i = 0, 49
	  !$omp task
	    !$omp cancellation point taskgroup
	    !$omp cancel taskgroup if (i.gt.5)
	  !$omp end task
	end do
      !$omp end taskgroup
    !$omp endsingle
  !$omp end parallel
end

File: libgomp/testsuite/libgomp.c-c++-common/cancel-taskgroup-2.c
/* { dg-do run } */
/* { dg-set-target-env-var OMP_CANCELLATION "true" } */

#include <stdlib.h>
#include <unistd.h>
#include <omp.h>

int
main ()
{
  #pragma omp parallel
  #pragma omp taskgroup
  #pragma omp task
  {
    #pragma omp cancel taskgroup
    if (omp_get_cancellation ())
      abort ();
  }
  #pragma omp parallel
  {
    #pragma omp barrier
    #pragma omp single
    #pragma omp taskgroup
    {
      int i;
      for (i = 0; i < 50; i++)
	#pragma omp task
	{
	  #pragma omp cancellation point taskgroup
	  usleep (30);
	  #pragma omp cancel taskgroup if (i > 5)
	}
    }
    usleep (10);
  }
  return 0;
}
